airflow-core/src/airflow/api_fastapi/auth/managers/base_auth_manager.py

```diff
@@ -347,6 +348,18 @@ def filter_authorized_menu_items(self, menu_items: list[MenuItem], *, user: T) -
         :param user: the user
         """
 
+    def is_authorized_hitl_task(self, user_id: str, assigned_users: Sequence[HITLUser]) -> bool:
```

Comment:
Let's try to follow the other APIs pattern. Also, I am not a big fan of having `HITLUser` as dependency of the base auth manager. In fact, we only need the user ID.

```suggestion
    def is_authorized_hitl_task(self, *, assigned_users: set[str], user: T) -> bool:
```